server/server.go

```diff
@@ -943,3 +948,68 @@ func isNotFound(err error) bool {
 
 	return false
 }
+
+// startReloadWatcherForMirrorRegistries starts a new goroutine that
+// monitors the "registries.conf.d" directory. If not set, it defaults to
+// "/etc/containers/registries.conf.d". The routine reloads the registries
+// configuration when a file is modified, created or deleted in the directory.
+func (s *Server) startReloadWatcherForMirrorRegistries(ctx context.Context, registriesConfDDir string) {
+	if registriesConfDDir == "" {
+		log.Infof(ctx, "No registries.conf.d directory specified, defaulting to /etc/containers/registries.conf.d")
+		registriesConfDDir = "/etc/containers/registries.conf.d"
+	}
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf(ctx, "Failed to create new watcher: %v", err)
+	}
+	defer watcher.Close()
+
+	log.Infof(ctx, "Registered reload watcher for mirror registries configuration")
+
+	s.reloadMirrorRegistriesConfiguration(ctx, watcher, registriesConfDDir)
```

Comment:
(It’s unclear to me what is the benefit of splitting this into two functions, and if the first one is “start reload watcher”, shouldn’t this one be “reload watcher”? “reload” is not quite an accurate name for the function operating on a watcher. See also elsewhere.)